binary-tree/src/lib.rs

```diff
@@ -51,4 +51,30 @@ mod tests {
             "85fc622076752a6fcda2c886c18058d639066a83473d9684704b5a29455ed2ed"
         )
     }
+
+    #[test]
+    fn test_merkleize_multiple_entries() {
+        let mut tree = Blake3Tree::new();
+        let mut keys = vec![
+            [0u8; KEY_LENGTH],
+            [0u8; KEY_LENGTH],
+            [0u8; KEY_LENGTH],
+            [0u8; KEY_LENGTH],
+        ];
+        keys[1][0] = 0x80;
+        keys[2][0] = 0x01;
+        keys[3][0] = 0x81;
+        for i in 0..keys.len() {
+            let mut new_array = [0u8; 32];
+            new_array[0] = i as u8;
```

Comment:
```suggestion
            new_array[0] = (i+1) as u8;
```
as specified in https://github.com/jsign/binary-tree-spec/blob/main/test_tree.py#L83 passes the test correctly with the original state root from the specs.